Dynatrace/dynatrace-operator
# Add token scope check for `apiToken` to troubleshoot command

# Description

`checkConnection` test has been moved to the `dynakube`  test group to keep `apiUrl` related tests in logical order.

New `checkDynatraceApiTokenScopes` test checks if tokens have required scopes based on the current DynaKube settings. Appropriate request is sent to the Dynatrace API (apiUrl) .

`troubleshoot` command provides now useful information if the ApiUrl is correct in terms of URL syntax but such a host/endpoint doesn't exist or tokens have missing scopes.

## How can this be tested?
Create `dynakube` secret with `apiToken` which doesn't have `InstallerDownload` scope. Check result
```
kubectl exec -n dynatrace -it `kubectl get -n dynatrace pods | grep operator | awk '{print $1}'` -- /usr/local/bin/dynatrace-operator troubleshoot
```


## Checklist
